File: Java/String/Anagram.java
//https://www.geeksforgeeks.org/problems/anagram-1587115620/1
if (s1.length() != s2.length()) {
            return false;
        }

        // Frequency array to store character counts
        int[] freq = new int[258];

        // Count frequencies for s1
        for (int i = 0; i < s1.length(); i++) {
            freq[s1.charAt(i)]++;
        }

        // Decrease frequencies for s2
        for (int i = 0; i < s2.length(); i++) {
            freq[s2.charAt(i)]--;
        }

        // Check if all frequencies are zero
        for (int i = 0; i < 258; i++) {
            if (freq[i] != 0) {
                return false;
            }
        }

        return true;

File: Python/String/Anagram.py
//https://www.geeksforgeeks.org/problems/anagram-1587115620/1
class Solution:
    # Function to check whether two strings are anagram of each other or not.
    def areAnagrams(self, s1, s2):
        # code here
        if len(s1) != len(s2):
            return False
        
        freq = [0] * 256  # To store frequency of characters
        for char in s1:
            freq[ord(char)] += 1
        for char in s2:
            freq[ord(char)] -= 1
        
        for count in freq:
            if count != 0:
                return False
        return True
